python/lsst/daf/persistence/posixStorage.py

```diff
@@ -863,7 +884,13 @@ def readYamlStorage(butlerLocation):
     for locationString in butlerLocation.getLocations():
         logLoc = LogicalLocation(butlerLocation.getStorage().locationWithRoot(locationString),
                                  butlerLocation.getAdditionalData())
-        finalItem = Policy(filePath=logLoc.locString())
+        if not os.path.exists(logLoc.locString()):
+            raise RuntimeError("No such YAML file: " + logLoc.locString())
```

Comment:
You missed the blue bar saying that the function had changed, I think.  I did add pickle-ability for PropertySet over in daf_base so that I could use the `getstate()` function in the YAML representer.

Reply:
I think  I was not clear. Why is there an error message "No such YAML file" in a function named `readPickleStorage`? What does YAML have to do with pickle storage?

I don't understand your statement about a blue bar. I see is that `readPickleStorage` has changed dramatically and the changes seem to have nothing to do with pickle format. For instance below you now check for `butlerLocation.pythonType == 'lsst.daf.persistence.RepositoryCfg'` which doesn't seem like a pickle, and if that test fails, then read as a yaml file. I'm really confused.

Reply:
`readPickleStorage` (starting at line 695) has not changed at all.  All these changes are in `readYamlStorage` (starting at line 870).